swe4103/swe4103-project
# Add TeamListView and corresponding unit tests

## 🛠️ What Type of PR is This?


## 📄 Description

This PR adds the TeamListView component, which displays classes, projects, and teams in a hierarchical tree structure. It also includes unit tests to validate the rendering of the structure and handles the case where the data is null, showing a fallback message.

## 🔗 Jira Ticket

[SCRUM-12](https://swe4103-pm-board.atlassian.net/browse/SCRUM-12)

## 🧾 Test Evidence

Test evidence includes the following successfully passing unit tests:

TeamListView renders correct structure: Validates that class, project, and team names are displayed as expected.
TeamListView null data handling: Confirms that a "No data available" message is shown when the data prop is null.
Testing steps:

Run **npm run test** to execute the unit tests.
Verify that all tests pass with no errors.

## 🧪 Added/Updated tests?

- [ ] No, and this is why: _Please r## 🛠️ What Type of PR is This?
